src/hook.c

```diff
@@ -204,7 +204,11 @@ static int hook_execute_va(git_buf *io, git_repository *repo, const char *name,
 
 	env.args.strings = (char **)git_vector_detach(&env.args.count, NULL, &arg_vector);
 
+	giterr_clear();
```

Comment:
We should probably check whether `git_vector_insert` succeeds or not. By ignoring its errors and clearing internal errors we may pass an invalid environment to the hook, which we should avoid.

Reply:
This ended up somehow in the -wip branch. As I can't see any valid reason to clear the error here, I'll 🔥 it.